File: TREES/Matrix/1333-sort-the-jumbled-numbers/sort-the-jumbled-numbers.py
class Solution:
    def sortJumbled(self, mapping: List[int], nums: List[int]) -> List[int]:
        pair = []
        for i, n in enumerate(nums):
            n = str(n)
            mapped_n = 0
            for c in n:
                mapped_n *= 10
                mapped_n += mapping[int(c)]
            pair.append((mapped_n, i))
        pair.sort()
        return [nums[p[1]] for p in pair]

File: TREES/Matrix/1333-sort-the-jumbled-numbers/sort-the-jumbled-numbers.cpp
class Solution {
public:
    vector<int> sortJumbled(vector<int>& mapping, vector<int>& nums) {
        vector<pair<int, int>> pairs; 
        for (int i = 0; i < nums.size(); ++i) {
            int n = nums[i];
            string s = to_string(n);
            int mapped_n = 0;
            
            // Map the digits of the number
            for (char c : s) {
                mapped_n *= 10;
                mapped_n += mapping[c - '0'];
            }
            
            // Store the mapped number and original index
            pairs.emplace_back(mapped_n, i);
        }
        sort(pairs.begin(), pairs.end());

        vector<int> result;
        for (const auto& p : pairs) {
            result.push_back(nums[p.second]);
        }
        return result;
    }
};